staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -118,17 +146,18 @@ func (w *watcher) Watch(ctx context.Context, key string, rev int64, opts storage
 	return wc, nil
 }
 
-func (w *watcher) createWatchChan(ctx context.Context, key string, rev int64, recursive, progressNotify bool, pred storage.SelectionPredicate) *watchChan {
+func (w *watcher) createWatchChan(ctx context.Context, key string, rev int64, recursive, progressNotify bool, initialEventsEndBookmarkRequired bool, pred storage.SelectionPredicate) *watchChan {
 	wc := &watchChan{
-		watcher:           w,
-		key:               key,
-		initialRev:        rev,
-		recursive:         recursive,
-		progressNotify:    progressNotify,
-		internalPred:      pred,
-		incomingEventChan: make(chan *event, incomingBufSize),
-		resultChan:        make(chan watch.Event, outgoingBufSize),
-		errChan:           make(chan error, 1),
+		watcher:                      w,
+		key:                          key,
+		initialRev:                   rev,
+		recursive:                    recursive,
+		progressNotify:               progressNotify,
+		internalPred:                 pred,
+		incomingEventChan:            make(chan *event, incomingBufSize),
+		resultChan:                   make(chan watch.Event, outgoingBufSize),
+		errChan:                      make(chan error, 1),
+		initialEventsEndBookmarkSent: !initialEventsEndBookmarkRequired,
```

Comment:
I don't think we really need to make it a field of watchChan.

I would rather make it a param of a `run()` method which is the only place where it is used.

Reply:
i'll have a look